### DISCUSSION TITLE
switching to windows-rs?

### DISCUSSION THREAD

**Discussion by LollipopFt** [gfx-rs/wgpu#2839] [General]
seeing as how progress on [winapi-rs](https://github.com/retep998/winapi-rs) has stalled, are there any thoughts on moving to [windows-rs](https://github.com/microsoft/windows-rs) especially since windows-rs allows you to pull more specific parts in compared to winapi-rs which may be a [concern](https://github.com/gfx-rs/wgpu/wiki/Code-Tree-Size)? thanks!

**Comment by jimblandy** [comment]
Firefox uses `wgpu` as its WebGPU implementation. For security reasons, Firefox doesn't download any sources from crates.io at build time: all the code except the Rust standard library is copied into the source tree using the `cargo vendor` command, to make it easier to detect and audit changes. But because windows-rs is so large, switching to it will have a substantial impact on the size of our source tree, and we'd like to put off switching to it for as long as we possibly can.

But obviously, just putting it off isn't going to solve anything. I'm pretty sure Firefox has a more solid plan - I just haven't been part of those discussions. I'll go talk to people and report back here.

**Comment by jimblandy** [comment]
Some Firefox discussion: https://bugzilla.mozilla.org/show_bug.cgi?id=1773509

It sounds like the plan is to carve out an exception for `windows-rs`, and work with upstream to reduce the size. (It's 200MB.)

**Comment by LollipopFt** [comment]
i see, thank you!
